lib/api_hammer/rails/check_required_params.rb

```diff
@@ -47,7 +47,7 @@ def check_required_params_helper(check, subparams, errors, parents)
           add_error.call(I18n.t(:"errors.required_params.must_be_type", :default => "%{key} must be a %{type}", :key => key, :type => check.name))
         end
       else
-        if subparams.is_a?(Hash)
+        if subparams.respond_to?(:to_h)
```

Comment:
can't use this here ! 
because `Array#to_h` is valid.
Also, this method is checking params with careful type check.
i dont think DuctTyping is appropriate in such a use case